components/brave_sync/bookmark_order_util_unittest.cc

```diff
@@ -80,18 +83,22 @@ TEST(BookmarkOrderUtilTest, GetOrder) {
 
   EXPECT_EQ(GetOrder("2.0.8", "2.0.8.1", ""), "2.0.8.0.1");
   EXPECT_EQ(GetOrder("2.0.8", "2.0.8.0.1", ""), "2.0.8.0.0.1");
+  EXPECT_EQ(GetOrder("2.0.8", "2.0.8.0.0.1", ""), "2.0.8.0.0.0.1");
 
   EXPECT_EQ(GetOrder("2.0.8.1", "2.0.9", ""), "2.0.8.2");
   EXPECT_EQ(GetOrder("2.0.8.1", "2.0.10", ""), "2.0.8.2");
   EXPECT_EQ(GetOrder("2.0.8.10", "2.0.8.15", ""), "2.0.8.11");
 
-  EXPECT_EQ(GetOrder("2.0.8.10", "2.0.8.15.1", ""), "2.0.8.10.1");
-  EXPECT_EQ(GetOrder("2.0.8.10", "2.0.8.11.1", ""), "2.0.8.10.1");
+  EXPECT_EQ(GetOrder("2.0.8.10", "2.0.8.15.1", ""), "2.0.8.11");
+  EXPECT_EQ(GetOrder("2.0.8.10", "2.0.8.11.1", ""), "2.0.8.11");
+
+  EXPECT_EQ(GetOrder("2.0.8.11", "2.0.8.11.1", ""), "2.0.8.11.0.1");
 
   EXPECT_EQ(GetOrder("2.0.8.10.0.1", "2.0.8.15.1", ""), "2.0.8.10.0.2");
```

Comment:
can we add a case `EXPECT_EQ(GetOrder("1.1.1.2.1", "1.1.1.3", ""), "1.1.1.2.2");`? This would happen when insert an item inside the end of a folder

Reply:
Yes, it is done in https://github.com/brave/brave-core/pull/2994/commits/035abe1219423064de303601a07039f88abd0844#diff-daae08bc23f88a92ad138a1e40c02ff5R111 .